### PR TITLE
Added CV2_Contrib, H5Py, Pandas, Pillow, PSUtil and SciPy 

### PR DESCRIPTION
**Pull request by peardox**
So actually it's six more (CV2_Contrib as an afterthought)

I've run some very basic tests and at least PSUtils does sensible things when told to...

```
  var t: Variant  := PSUtil.psutil.cpu_count();
  Log('api returned ' + t);

```

In my case returning 16... which is correct (8c/16t)

Spotted a bug in earlier version of CV2_Contrib using np as module name - it displayed cv2 as the PyModuleName but I'm cutting + pasting to make these an missed the fact that Numpy declared the property np - changing this to cv2 etc made everything work as expected (so I assume the untested yet ones are OK)

All the modules included in this bunch are important.

Torch (and other things) import Pillow anyway but it made sense to also have a component for it so I threw that one together as well

Everything installs in the IDE under the 'Python - Data Science Ecosystem' branch

I'll go add a note to the issue I fixed recently where I had the wrong variant (duhh)

### CODE REVIEW

**Comment by peardox** [comment]
Just tested that issue about cv2 with CV2_Contrib - works nicely (using cv2 as property - previous bug in my component fixed so property properly named)

**Comment by lmbelo** [comment]
@peardox thank you for your contribution. I'm only finishing some tasks before reviewing and accepting it.

**Comment by peardox** [comment]
I've tested the components I could write quick tests for as outlined above

Doubt there will be any problems as the things are simple enough to start with. I will be trying Pillow out soon though...

**Comment by lmbelo** [comment]
> I've tested the components I could write quick tests for as outlined above
> 
> Doubt there will be any problems as the things are simple enough to start with. I will be trying Pillow out soon though...

Can we create a test project covering up all these packages? We can get started with simple module importation and basic stuff.

**Comment by peardox** [comment]
I was going to start by writing a Tutorial showing how to install all the libraries then illustrate use with 'HelloCPU' which would use PSUtil to show machine status (CPU cores/ threads/ memory/ VM and that kinda thing) with both Python and Module versions of the code to show that you can almost re-write the python in Delphi.

After that I was thinking of Pillow to demo some basic Image stuff

The problem is that the libraries all do very different things so while you might come up with a way to use several in a constructive way using all of them would be pure demo for demo sake (which is actually also worthwhile...)

My main project uses a load of libraries and will be open source (it also use Skia though...)

**Comment by lmbelo** [comment]
Let's break it down to separated PR's with smaller commits in future requests, it gets easier to merge.

File: packages/P4DCV2_Contrib.dpk [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package P4DCV2_Contrib;
```
Comment (lmbelo) [inline packages/P4DCV2_Contrib.dpk:1]: Let's rename it to P4DCV2Contrib.

File: packages/dclP4DCV2_Contrib.dpk [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package dclP4DCV2_Contrib;
```
Comment (lmbelo) [inline packages/dclP4DCV2_Contrib.dpk:1]: Let's rename it to dclP4DCV2Contrib.

File: src/CV2_Contrib/CV2_Contrib.pas [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ *                                                      *
+ * Copyright (c) 2022 by Embarcadero Technologies       *
+ * Licensed under the MIT License                       *
+ *                                                      *
+ * For full license text and more information visit:    *
+ * https://github.com/Embarcadero/P4D-Data-Sciences     *
+ ********************************************************)
+
+
+unit CV2_Contrib;
```
Comment (lmbelo) [inline src/CV2_Contrib/CV2_Contrib.pas:12]: Let's rename it to CV2Contrib.

File: src/CV2_Contrib/CV2_ContribReg.pas [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ *                                                      *
+ * Copyright (c) 2022 by Embarcadero Technologies       *
+ * Licensed under the MIT License                       *
+ *                                                      *
+ * For full license text and more information visit:    *
+ * https://github.com/Embarcadero/P4D-Data-Sciences     *
+ ********************************************************)
+
+
+unit CV2_ContribReg;
```
Comment (lmbelo) [inline src/CV2_Contrib/CV2_ContribReg.pas:12]: Let's rename it to CV2ContribReg.

File: packages/P4DMatplotLib.dproj [hunk 5]
```diff
@@ -5,10 +5,10 @@
         <Config Condition="'$(Config)'==''">Release</Config>
         <FrameworkType>None</FrameworkType>
         <MainSource>P4DMatplotLib.dpk</MainSource>
-        <Platform Condition="'$(Platform)'==''">Win32</Platform>
```
Comment (lmbelo) [inline packages/P4DMatplotLib.dproj:8]: What is the purpose of these changes? Should we keep it?

File: packages/P4DNLTK.dproj [hunk 6]
```diff
@@ -5,10 +5,10 @@
         <Config Condition="'$(Config)'==''">Release</Config>
         <FrameworkType>None</FrameworkType>
         <MainSource>P4DNLTK.dpk</MainSource>
-        <Platform Condition="'$(Platform)'==''">Win32</Platform>
```
Comment (lmbelo) [inline packages/P4DNLTK.dproj:8]: What is the purpose of these changes? Should we keep it?

File: packages/P4DNumPy.dproj [hunk 7]
```diff
@@ -5,10 +5,10 @@
         <Config Condition="'$(Config)'==''">Release</Config>
         <FrameworkType>None</FrameworkType>
         <MainSource>P4DNumPy.dpk</MainSource>
-        <Platform Condition="'$(Platform)'==''">Win32</Platform>
```
Comment (lmbelo) [inline packages/P4DNumPy.dproj:8]: What is the purpose of these changes? Should we keep it?

File: packages/P4DOpenCV.dproj [hunk 8]
```diff
@@ -5,10 +5,10 @@
         <Config Condition="'$(Config)'==''">Release</Config>
         <FrameworkType>None</FrameworkType>
         <MainSource>P4DOpenCV.dpk</MainSource>
-        <Platform Condition="'$(Platform)'==''">Win32</Platform>
```
Comment (lmbelo) [inline packages/P4DOpenCV.dproj:8]: What is the purpose of these changes? Should we keep it?

File: packages/P4DScikitLearn.dproj [hunk 9]
```diff
@@ -5,10 +5,10 @@
         <Config Condition="'$(Config)'==''">Release</Config>
         <FrameworkType>None</FrameworkType>
         <MainSource>P4DScikitLearn.dpk</MainSource>
-        <Platform Condition="'$(Platform)'==''">Win32</Platform>
```
Comment (lmbelo) [inline packages/P4DScikitLearn.dproj:8]: What is the purpose of these changes? Should we keep it?

File: packages/P4DTensorFlow.dproj [hunk 10]
```diff
@@ -5,10 +5,10 @@
         <Config Condition="'$(Config)'==''">Release</Config>
         <FrameworkType>None</FrameworkType>
         <MainSource>P4DTensorFlow.dpk</MainSource>
-        <Platform Condition="'$(Platform)'==''">Win32</Platform>
```
Comment (lmbelo) [inline packages/P4DTensorFlow.dproj:8]: What is the purpose of these changes? Should we keep it?

File: packages/dclP4DMatplotLib.dproj [hunk 11]
```diff
@@ -7,12 +7,17 @@
         <MainSource>dclP4DMatplotLib.dpk</MainSource>
         <Platform Condition="'$(Platform)'==''">Win32</Platform>
         <ProjectGuid>{B53E1A32-9471-4362-B9D7-60FF0E17FC89}</ProjectGuid>
-        <ProjectVersion>19.4</ProjectVersion>
```
Comment (lmbelo) [inline packages/dclP4DMatplotLib.dproj:10]: What is the purpose of these changes? Should we keep it?

File: packages/dclP4DNLTK.dproj [hunk 12]
```diff
@@ -7,12 +7,17 @@
         <MainSource>dclP4DNLTK.dpk</MainSource>
         <Platform Condition="'$(Platform)'==''">Win32</Platform>
         <ProjectGuid>{B47E9E59-70B7-46BF-8790-244D93DC224F}</ProjectGuid>
-        <ProjectVersion>19.4</ProjectVersion>
```
Comment (lmbelo) [inline packages/dclP4DNLTK.dproj:10]: What is the purpose of these changes? Should we keep it?

File: packages/dclP4DNumPy.dproj [hunk 13]
```diff
@@ -7,12 +7,17 @@
         <MainSource>dclP4DNumPy.dpk</MainSource>
         <Platform Condition="'$(Platform)'==''">Win32</Platform>
         <ProjectGuid>{E3059032-7656-430C-89A2-B389B2151E88}</ProjectGuid>
-        <ProjectVersion>19.4</ProjectVersion>
```
Comment (lmbelo) [inline packages/dclP4DNumPy.dproj:10]: What is the purpose of these changes? Should we keep it?

File: packages/dclP4DOpenCV.dproj [hunk 14]
```diff
@@ -7,12 +7,17 @@
         <MainSource>dclP4DOpenCV.dpk</MainSource>
         <Platform Condition="'$(Platform)'==''">Win32</Platform>
         <ProjectGuid>{06D62157-0E57-4940-925B-715687CBF0AD}</ProjectGuid>
-        <ProjectVersion>19.4</ProjectVersion>
```
Comment (lmbelo) [inline packages/dclP4DOpenCV.dproj:10]: What is the purpose of these changes? Should we keep it?

File: packages/dclP4DPyTorch.dproj [hunk 15]
```diff
@@ -7,12 +7,17 @@
         <MainSource>dclP4DPyTorch.dpk</MainSource>
         <Platform Condition="'$(Platform)'==''">Win32</Platform>
         <ProjectGuid>{EE4B450F-7C20-400E-8149-C01F88A3F480}</ProjectGuid>
-        <ProjectVersion>19.4</ProjectVersion>
```
Comment (lmbelo) [inline packages/dclP4DPyTorch.dproj:10]: What is the purpose of these changes? Should we keep it?

File: packages/dclP4DScikitLearn.dproj [hunk 16]
```diff
@@ -7,12 +7,17 @@
         <MainSource>dclP4DScikitLearn.dpk</MainSource>
         <Platform Condition="'$(Platform)'==''">Win32</Platform>
         <ProjectGuid>{B8B30567-EF48-4563-88DD-CCC56D22DF4F}</ProjectGuid>
-        <ProjectVersion>19.4</ProjectVersion>
```
Comment (lmbelo) [inline packages/dclP4DScikitLearn.dproj:10]: What is the purpose of these changes? Should we keep it?

File: packages/dclP4DTensorFlow.dproj [hunk 17]
```diff
@@ -7,12 +7,17 @@
         <MainSource>dclP4DTensorFlow.dpk</MainSource>
         <Platform Condition="'$(Platform)'==''">Win32</Platform>
         <ProjectGuid>{CCFDA41F-5004-4E54-997D-5967D19564F7}</ProjectGuid>
-        <ProjectVersion>19.4</ProjectVersion>
```
Comment (lmbelo) [inline packages/dclP4DTensorFlow.dproj:10]: What is the purpose of these changes? Should we keep it?

Comment (peardox) [reply packages/dclP4DTensorFlow.dproj:10]: May have been CE making numbering changes although I did have to alter a library version in another PR

Comment (lmbelo) [reply packages/dclP4DTensorFlow.dproj:10]: We can avoid these .dproj changes in existing packages.

Comment (peardox) [reply packages/dclP4DTensorFlow.dproj:10]: Dunno, may affect backward compatibility?

Not sure but thern again not had any problems with them myelf

Comment (lmbelo) [reply packages/dclP4DTensorFlow.dproj:10]: We are using Win32 Release as the default target platform and build configuration. It has changed in the .dproj files you've shared.

Comment (lmbelo) [reply packages/dclP4DTensorFlow.dproj:10]: I'm merging it up and doing the adjustments myself. You did it great, thanks! 😄

**Comment by peardox** [comment]
I've just done CV2Contrib but not tested yet

**Comment by lmbelo** [comment]
> I've just done CV2Contrib but not tested yet



> I've just done CV2Contrib but not tested yet

Do you want to update this pull request with it? If possible, remove the existing .dproj files.
